src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py

```diff
@@ -0,0 +1,208 @@
+        """Prepare roles"""
+        logger.info("Preparing roles in chat group.")
+        # check roles is a non-empty list of ChatRole
+        if not isinstance(roles, list) or len(roles) == 0 or not all(isinstance(role, ChatRole) for role in roles):
+            raise ChatGroupError(f"Agents should be a non-empty list of ChatRole. Got {roles!r} instead.")
+
+        # check entry_role is in roles
+        if entry_role is not None and entry_role not in roles:
+            raise ChatGroupError(f"Entry role {entry_role.name} is not in roles list {roles!r}.")
+
+        speak_order_list = self._get_speak_order(roles, entry_role, speak_order)
+        roles_dict = {role.name: role for role in roles}
+        return roles_dict, cycle(speak_order_list)
+
+    def _get_speak_order(
+        self, roles: List[ChatRole], entry_role: Optional[ChatRole], speak_order: ChatGroupSpeakOrder
+    ) -> List[str]:
+        """Calculate speak order"""
+        if speak_order == ChatGroupSpeakOrder.SEQUENTIAL:
+            if entry_role:
+                logger.warn(
+                    f"Entry role {entry_role.name!r} is ignored when speak order is sequential. "
+                    f"The first role in the list will be the entry role: {roles[0].name!r}."
+                )
+
+            speak_order_list = [role.name for role in roles]
+            logger.info(f"Role speak order is {speak_order_list!r}.")
+            return speak_order_list
+        else:
+            raise NotImplementedError(f"Speak order {speak_order.value!r} is not supported yet.")
+
+    @staticmethod
+    def _validate_int_parameters(max_turns: int, max_tokens: int, max_time: int):
+        """Validate int parameters"""
+        logger.debug("Validating integer parameters for chat group.")
+        if max_turns is not None and not isinstance(max_turns, int):
+            raise ChatGroupError(f"max_turns should be an integer. Got {type(max_turns)!r} instead.")
+        if max_tokens is not None and not isinstance(max_tokens, int):
+            raise ChatGroupError(f"max_tokens should be an integer. Got {type(max_tokens)!r} instead.")
+        if max_time is not None and not isinstance(max_time, int):
+            raise ChatGroupError(f"max_time should be an integer. Got {type(max_time)!r} instead.")
+
+        logger.info(
+            f"Chat group maximum turns: {max_turns!r}, maximum tokens: {max_tokens!r}, maximum time: {max_time!r}."
+        )
+        return max_turns, max_tokens, max_time
+
+    def invoke(self, *args, **kwargs):
+        """Invoke the chat group"""
+        if args or kwargs:
+            logger.warn(
+                f"Chat group invoke does not accept arguments, got {args!r} and {kwargs!r} instead and ignored them."
+            )
+
+        logger.info("Invoking chat group.")
+
+        chat_round = 0
+        chat_token = 0
+        chat_start_time = time.time()
+        while True:
```

Comment:
make sure we have good trace for the orchestration process

Reply:
Do you mean the log details? This is the current orchestration log if log level is set to INFO:
![image](https://github.com/microsoft/promptflow/assets/2572521/3847431a-3c7a-4b83-ba9c-b14b62a492e1)

Reply:
Offline synced, added an item to track: 3015443